### PR TITLE
unset_bone_position

### PR DESCRIPTION
**Pull request by theviper121**
Add compact, short information about your PR for easier understanding:

- Add function to restore a bone to original position and animation after setting position with set_bone_position
- Adds unset_bone_position(bone) function
- unsure
- Allows more complex player animations like swimming by rotating the player model using set_bone_position and being able to reset afterwards.

This PR is **ready for review**.

Set a bone position, then unset it and confirm bone is in original position.

### CODE REVIEW

File: src/activeobject.h [hunk 1]
```diff
@@ -64,6 +64,7 @@ enum ActiveObjectCommand {
 	AO_CMD_UPDATE_ARMOR_GROUPS,
 	AO_CMD_SET_ANIMATION,
 	AO_CMD_SET_BONE_POSITION,
+	AO_CMD_UNSET_BONE_POSITION,
```
Comment (sfan5) [inline src/activeobject.h:67]: Adding something in the middle of this enum will change the values, breaking backwards compatibility.

**Comment by Zughy** [comment]
Shouldn't be better `reset_bone_position` instead of `unset`? Also, isn't already a way to do that? I'm genuinely asking, as I'm no bones expert

Lastly, documentation is missing

**Comment by theviper121** [comment]
There is no way to reset a bone prior to this pr. I feel the name of the function is very suitable. I feel reset_bone_position could be interpreted differently.

**Comment by hecktest** [comment]
I agree that reset_bone_position could imply resetting it to the bind pose. How about `free_bone` or `unbind_bone`?

File: src/server/unit_sao.cpp [hunk 2]
```diff
@@ -112,6 +122,12 @@ void UnitSAO::sendOutdatedData()
 			m_messages_out.emplace(getId(), true, generateUpdateBonePositionCommand(
 				bone_pos.first, bone_pos.second.X, bone_pos.second.Y));
 		}
+		for (const auto &bone_pos : m_bone_position_unset) {
+			if(!m_bone_position_unset[bone_pos.first]) {
+				m_messages_out.emplace(getId(), true, generateUpdateBonePositionUnsetCommand(bone_pos.first));
+				m_bone_position_unset[bone_pos.first] = true;
+			}
+		}
```
Comment (sfan5) [inline src/server/unit_sao.cpp:130]: I don't see why you have to store a boolean for this.
You could do use a list, add the bone name when it is to be removed (in `unsetBonePosition`) and clear the list here after generating all commands.

**Comment by Desour** [comment]
#12388

**Pull request closed without merging**